include/wx/msw/srchctrl.h

```diff
@@ -0,0 +1,123 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        wx/msw/srchctrl.h
+// Purpose:     wxSearchCtrl class
+// Author:      Kettab Ali
+// Created:     2019-12-23
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_MSW_SEARCHCTRL_H_
+#define _WX_MSW_SEARCHCTRL_H_
+
+#if wxUSE_SEARCHCTRL
+
+// ----------------------------------------------------------------------------
+// wxSearchCtrl
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxSearchCtrl : public wxSearchCtrlBase
+{
+public:
+    // creation
+    // --------
+
+    wxSearchCtrl() : wxSearchCtrlBase()
+    {
+        Init();
+    }
+
+    wxSearchCtrl(wxWindow *parent, wxWindowID id,
+               const wxString& value = wxEmptyString,
+               const wxPoint& pos = wxDefaultPosition,
+               const wxSize& size = wxDefaultSize,
+               long style = 0,
+               const wxValidator& validator = wxDefaultValidator,
+               const wxString& name = wxSearchCtrlNameStr)
+        : wxSearchCtrlBase()
+    {
+        Init();
+        Create(parent, id, value, pos, size, style, validator, name);
+    }
+
+    virtual ~wxSearchCtrl();
+
+    bool Create(wxWindow *parent, wxWindowID id,
+                const wxString& value = wxEmptyString,
+                const wxPoint& pos = wxDefaultPosition,
+                const wxSize& size = wxDefaultSize,
+                long style = 0,
+                const wxValidator& validator = wxDefaultValidator,
+                const wxString& name = wxSearchCtrlNameStr);
+
+#if wxUSE_MENUS
+    // get/set search button menu
+    // --------------------------
+    virtual void SetMenu( wxMenu* menu ) wxOVERRIDE;
+    virtual wxMenu* GetMenu() wxOVERRIDE { return m_menu; }
+#endif // wxUSE_MENUS
+
+    // get/set search options
+    // ----------------------
+    virtual void ShowSearchButton( bool show ) wxOVERRIDE;
+    virtual bool IsSearchButtonVisible() const wxOVERRIDE;
+
+    virtual void ShowCancelButton( bool show ) wxOVERRIDE;
+    virtual bool IsCancelButtonVisible() const wxOVERRIDE;
+
+    virtual void SetDescriptiveText(const wxString& text) wxOVERRIDE;
+    virtual wxString GetDescriptiveText() const wxOVERRIDE;
+
+#if wxUSE_MENUS
+    void PopupSearchMenu();
```

Comment:
I'm not really sure if this should be a public function. IMO public functions should be either documented (which this one isn't) or explicitly marked as internal (and ideally use `WX` prefix in this case to avoid name clashes). So the simplest would be to just remove it (i.e. remove it and inline its code into the function from which it is called).